share/p2p/peers/manager.go

```diff
@@ -210,23 +230,33 @@ func (s *Manager) subscribeHeader(ctx context.Context, headerSub libhead.Subscri
 			continue
 		}
 
-		s.getOrCreatePool(h.DataHash.String()).markValidated()
+		if s.getOrCreatePool(h.DataHash.String()).markValidated() {
+			log.Debugw("marked validated", "datahash", h.DataHash.String())
+		}
```

Comment:
technically if we're calling `markValidated` from here, then it's always true no? so we always log - is there a reason to do the bool check ?

Reply:
No, markValidated will return true only if it actually swapped `isValidated` flag. If pool was validated before, it will return false and no log line should appear